optimizer-hints.md

```diff
@@ -814,3 +786,78 @@ SELECT /*+ NTH_PLAN(3) */ count(*) from t where a > 5;
+| Level   | Code | Message                                                              |
++---------+------+----------------------------------------------------------------------+
+| Warning | 1815 | use_index(.t, a) is inapplicable, check whether the table(.t) exists |
++---------+------+----------------------------------------------------------------------+
+1 row in set (0.00 sec)
+```
+
+由于无法识别表 `t` 对应的数据库名，因此 `use_index(t, a)` 无法生效。
+
+### 跨库查询不指定库名导致 Hint 不生效
+
+对于跨库查询中需要访问的表，需要显示的指定库名，否则可能出现 Hint 失效的情况。例如：
+
+```sql
+mysql> use test1;
+Database changed
+mysql> create table t1(a int, key(a));
+Query OK, 0 rows affected (0.02 sec)
+
+mysql> use test2;
+Database changed
+mysql> create table t2(a int, key(a));
+Query OK, 0 rows affected (0.01 sec)
+
+mysql> select /*+ use_index(t1, a) */ * from test1.t1, t2;
+Empty set, 1 warning (0.00 sec)
+
+mysql> show warnings;
++---------+------+----------------------------------------------------------------------------------+
+| Level   | Code | Message                                                                          |
++---------+------+----------------------------------------------------------------------------------+
+| Warning | 1815 | use_index(test2.t1, a) is inapplicable, check whether the table(test2.t1) exists |
++---------+------+----------------------------------------------------------------------------------+
+1 row in set (0.00 sec)
+```
+
+由于 `t1` 不在当前数据库 `test2` 下，因此 Hint `use_index(t1, a)` 无法正确的识别，需要显示加上库名 `use_index(test1.t1, a)`。
+
+### Hint 位置不正确导致不生效
+
+如果 Hint 的位置没有跟在指定的关键字后面，会无法生效。例如：
+
+```sql
+mysql> select * /*+ use_index(t, a) */ from t;
+Empty set, 1 warning (0.01 sec)
+
+mysql> show warnings;
++---------+------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| Level   | Code | Message                                                                                                                                                                                                                 |
++---------+------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| Warning | 1064 | You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use [parser:8066]Optimizer hint can only be followed by certain keywords like SELECT, INSERT, etc. |
++---------+------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+1 row in set (0.01 sec)
+```
```

Comment:
````suggestion
如果没有将 Hint 正确地放在指定关键字的后面，它将无法生效。例如：

```sql
SELECT * /*+ use_index(t, a) */ FROM t;
SHOW WARNINGS;
```

Warning 信息如下：

```sql
+---------+------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Level   | Code | Message                                                                                                                                                                                                                 |
+---------+------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Warning | 1064 | You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use [parser:8066]Optimizer hint can only be followed by certain keywords like SELECT, INSERT, etc. |
+---------+------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.01 sec)
```
````